### PR TITLE
Add tests for api requests

### PR DESCRIPTION
**Pull request by akwong372**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const app = express();
+const db = require('./db.js');
+
+app.use(express.static('./public/dist'));
+
+app.get('/api/users', (req, res) => {
+  db.User.findAll()
```
Comment (LesliePajuelo) [inline server/app.js:8]: Set a limit on how many records can be returned from a findAll()

File: server/app.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    .then((users) => {
+      res.send(users);
+    });
```
Comment (LesliePajuelo) [inline server/app.js:11]: And if there are no users? What does sequalize return? null, 0, undefined, error?

File: server/server.js [hunk 3]
```diff
@@ -1,28 +1,6 @@
-const express = require('express');
-const app = express();
-const db = require('./db.js');
+const app = require('./app.js');
 const port = 1337;
```
Comment (LesliePajuelo) [inline server/server.js]: Bold choice

Comment (LesliePajuelo) [inline server/server.js:2]: Bold choice for a port number

File: tests/api.test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+const db = require('../server/db.js');
+const app = require('../server/app.js');
+
+afterAll(() => db.sequelize.close());
+
+describe('Api should get data from database', () => {
+  test('It should respond with an array of user objects', () => {
+    return request(app).get('/api/users')
+      .then(response => {
+        expect(response.body.length).toBe(50);
```
Comment (LesliePajuelo) [inline tests/api.test.js:11]: Hard coding to a number of records in a database is not recommended unless you're clearing and reseeding the db before each test.

File: tests/api.test.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    return request(app).get('/api/restaurants/UeS2aZ7ryI/reviews')
+      .then(response => {
+        expect(response.body.length).toBeGreaterThan(0);
+      });
+  });
+
+  test('It should handle pages that do not exist', () => {
+    return request(app).get('/doesntexist')
+      .then(response => {
+        expect(response.error.text).toBe(`Use endpoints '/api/users', or '/api/restaurants/[restaurant ID]/reviews', or '/api/restaurants/reviews'`);
```
Comment (LesliePajuelo) [inline tests/api.test.js:32]: Testing for specific text is also an antipattern. Simply test that the the error.text exists.

File: server/app.js (hunk 2 above)
Comment (akwong372) [reply server/app.js:11]: It should be returning null

File: server/server.js (hunk 3 above)
Comment (akwong372) [reply server/server.js:2]: It's what I aspire to be

**Pull request merged**